configuration/external_parameters.rst

```diff
@@ -99,6 +99,86 @@ will be used whenever the corresponding environment variable is *not* found:
         // config/services.php
         $container->setParameter('env(DATABASE_HOST)', 'localhost');
 
+Environment Variable Processors
+-------------------------------
+
+When using environment variables they are always strings by default, but sometimes
+you will want to have specific types so that they match the types expected by your code.
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/framework.yaml
+        framework:
+            router:
+                http_port: env(int:HTTP_PORT)
+
+    .. code-block:: xml
+
+        <!-- config/packages/framework.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:framework="http://symfony.com/schema/dic/symfony"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/symfony
+                http://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+
+            <framework:config>
+                <framework:router http_port="%env(int:HTTP_PORT)%" />
+            </framework:config>
+        </container>
+
+    .. code-block:: php
+
+        // config/packages/doctrine.php
+        $container->loadFromExtension('framework', array(
+            'router' => array(
+                'http_port' => '%env(int:HTTP_PORT)%',
+            )
+        ));
+
+A number of different types are supported:
+
+``env(string):FOO)``
```

Comment:
I'm not a big fan of using `foo` and `bar` in docs because I prefer to use more realistic examples. For example we could use `ENV_VAR_NAME` instead of `FOO`. But before making any change, let's wait for other opinions.

Reply:
I think its a fair point, and more consistent with the style of the docs in general